packages/react-router/src/navigationItemRegistry.ts

```diff
@@ -0,0 +1,30 @@
+import type { LinkProps } from "react-router-dom";
+import type { ReactNode } from "react";
+
+export type NavigationItem = Omit<LinkProps, "children"> & {
+    content: ReactNode;
+    children?: NavigationItem[];
+    additionalProps?: Record<string, unknown>;
+};
+
+export type RootNavigationItem = NavigationItem & {
+    // Highest priority is rendered first.
+    priority?: number;
+};
+
+export class NavigationItemRegistry {
+    private _items: RootNavigationItem[];
+
+    constructor() {
+        this._items = [];
+    }
+
+    add(navigationItems: RootNavigationItem[]) {
+        // Create a new array so the navigation items array is immutable.
+        this._items = [...this._items, ...navigationItems.filter(x => x)];
```

Comment:
...navigationItems.filter(x => x) feels odd. I understand you want to filter ou NavigationItems that are undefined. However, according to the typings, undefined is not allowed in the array. So if we feel like they still might be undefined, i would probably do the validation when first loading the navigation items, not here

Reply:
Might be considered as premature optimisation but I feel like it's better to filter the undefined items once when adding them than on every re-render.

What the issue with filtering the items here? I it's was a "get" function, I would not feel good about it either but since it's a "command" function it seems legit.

Reply:
Ah got it, so what you are saying is that it's impossible to have undefined items in the array with the current typings?

Reply:
yes!